pkg/agent/controller/traceflow/packetin.go

```diff
@@ -161,8 +161,13 @@ func (c *Controller) parsePacketIn(pktIn *ofctrl.PacketIn) (*crdv1alpha1.Tracefl
 	}
 
 	// Collect Service DNAT and SNAT.
+	// The packet of Service is DNATed only, ipDst != ctNwDst.
+	// The packet of Service is both DNATed and SNATed, ipSrc is Antrea gateway IP.
+	gatewayIPv4 := c.nodeConfig.GatewayConfig.IPv4
+	gatewayIPv6 := c.nodeConfig.GatewayConfig.IPv6
 	if !tfState.receiverOnly {
-		if isValidCtNw(ctNwDst) && ipDst != ctNwDst {
+		if isValidCtNw(ctNwDst) && ipDst != ctNwDst ||
```

Comment:
For Service traffic, DNAT is a must, so the 2nd condition is possibly never verified (`isValidCtNw(ctNwSrc) && ipSrc != ctNwSrc && (gatewayIPv4 != nil && ipSrc == gatewayIPv4.String() || gatewayIPv6 != nil && ipSrc == gatewayIPv6.String()`)

Reply:
How about this? This method can also work for Windows.
```
		// ctNwDst is same as ipDst in Windows, use originalIPDst as an alternative.
		originalIPDst := ""
		if tfState.packet != nil {
			originalIPDst = tfState.packet.DestinationIP.String()
		}
		if isValidCtNw(ctNwDst) && ipDst != ctNwDst || (originalIPDst != "" && originalIPDst != ipDst) {
```

Reply:
> For Service traffic, DNAT is a must, so the 2nd condition is possibly never verified (`isValidCtNw(ctNwSrc) && ipSrc != ctNwSrc && (gatewayIPv4 != nil && ipSrc == gatewayIPv4.String() || gatewayIPv6 != nil && ipSrc == gatewayIPv6.String()`)

@wenyingd If a packet passes SNAT CT zone later than DNAT CT zone, the ctNwDst is same as ipDst since its information is lost.